File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        
        int[] arr = new int[2];
        int temp = 0;
        int cnt = 0;
        int num = 0;
        
        for(int i : lottos) {
            for(int c : win_nums) {
                if (i == c) {
                    cnt++;
                }
            }
            if (i == 0){
                temp++;
            }
        }
        
        for (int i = 6; i > 0; i--) {
            num++;
            if (temp + cnt == 0) {
                arr[0] = 6;
                arr[1] = 6;
                break;
            }
            
            
            if (temp + cnt == i) {
                arr[0] = num;
            }
            if (cnt == i) {
                arr[1] = num;
            }
            if (cnt == 0) {
                arr[1] = 6;
                arr[0] = 1;
            }
        }
        return arr;
    }
}

File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    count = 0
    for i in lottos:
        if i in win_nums:
            count += 1
    answer.append(7-count-lottos.count(0) if 0 in lottos or count != 0 else 6)
    answer.append(6 if count == 0 else 7-count)
    return answer